### DISCUSSION TITLE
How to print out a text file's contents?

### DISCUSSION THREAD

**Discussion by Mayo374** [CosmosOS/Cosmos#2418] [Q&A]
Hey there I'm currently working on my brand new Update for my operating system. I've been implementing a lot of features into it recently. One of those features was a notepad. Got it to the point where it can make a text file and write stuff to the file I just need to know how to print out the contents of the text file. Thank you 👍

**Comment by AsertCreator** [comment]
firstly its not a bug, secondly you need to make sure that requested file exists and then use streamreader and file stream to get file contents

**Comment by LoxyMono** [comment]
I am 99.9% sure there is a given method in the wiki here
https://github.com/CosmosOS/Cosmos/wiki/FAT-File-System look downwards, it should show an example

**Comment by Mayo374** [comment]
> I am 99.9% sure there is a given method in the wiki here https://github.com/CosmosOS/Cosmos/wiki/FAT-File-System look downwards, it should show an example

Oh wow I didn't even notice that cosmos had a documentation. Thank you very much!

**Comment by Mayo374** [comment]
> firstly its not a bug, secondly you need to make sure that requested file exists and then use stream reader and file stream to get file contents

I know its not a bug but there wasn't a help wanted tag so yea. I tried streamreader and it doesn't work. Whenever I try and print out the contents of streamreader it just crashes the OS.

**Comment by retrievaaaa** [comment]
> > firstly its not a bug, secondly you need to make sure that requested file exists and then use stream reader and file stream to get file contents
> 
> I know its not a bug but there wasn't a help wanted tag so yea. I tried streamreader and it doesn't work. Whenever I try and print out the contents of streamreader it just crashes the OS.

Thats what discussions are for 🙂 
https://github.com/CosmosOS/Cosmos/discussions

**Comment by winbamstudios** [comment]
**move this to discussions**
